Add flag to specify path to config, update config to support binding to a specific address

opalmer opened the pull request:
This PR adds a command line flag to specify the path to a json file instead of assuming the path will in the current directory. It also updates the json configuration file to support an alternate address to bind to bind to. This change shouldn't impact any existing configurations or deployments.

kaakaa commented on .gitignore:
```diff
@@ -1,4 +1,5 @@
-coverage.txt
\ No newline at end of file
+coverage.txt
+/matterpoll-emoji
```
Sorry, I could not understand why this line is added.

opalmer replied:
I added this because if you run `go build .` you get a binary named `matterpoll-emoji` which shouldn't be checked into the repository since it's a build artifact and is platform/architecture specific. I don't install go on production servers so I have to build a binary which is probably going to be more common than using Go to run main.go so I thought it was a logical change for most people.

kaakaa replied:
Okay, I got it!
We need this line, Thank you!

opalmer commented:
@kaakaa thanks for the review! Latest commit should have the updates you requested. I also fixed a couple of other issues:

* Fixed comments in the json example in the readme. Technically speaking json does not support comments but `//` is a valid comment in JavaScript and certain tools do know how to remove comments from json files.
* Added `json:"<field>"` to the configuration structs. While not required it makes it more explicit and more apparent that those structs are intended to be populated from json data.

Hanzei commented:
Hey @opalmer 
What do you think about combining the `port` and `address` setting? They are quite similar and it would reduce the number of options.

opalmer commented:
I actually thought about doing that initially but didn't because it wouldn't be backwards compatible. If you're ok with that, I'm happy to make the change.

Hanzei commented:
The port setting was added today, so I don't mind😃

opalmer commented:
Haha, good point. Fixed :).

kaakaa commented on main.go:
```diff
@@ -1,23 +1,26 @@
 )
 
+var config = flag.String(
+	"c", "config.json", "optional path to the config file")
```
Do we need to separate line here?

opalmer replied:
Done.

Hanzei commented on .config.json:
```diff
@@ -1,9 +1,9 @@
-		"password": "botbot"
-	}
+  "host": "http://mattermost.example.com:8065",
+  "listen": "localhost:8505",
```
Shouldn't this be `:8505`?

opalmer replied:
It could be either but I left it this way because it's more secure and I imagine most will have Mattermost and this project running on the same server. The problem with `:8505` is it will listen on all interfaces which means you could be listening on an interface which is facing the internet or another network segment unintentionally. There's too many possible scenarios where that could be a problem so it's best to air on the side of caution in my opinion.

Hanzei replied:
:+1:

Hanzei commented on README.md:
```diff
@@ -37,14 +37,15 @@ cp .config.json config.json
-   "id": "bot",          # The username of an existing Mattermost account
-   "password": "botbot"  # The password of an existing Mattermost account
+   "id": "bot",  // The username of an existing Mattermost account
+   "password": "botbot"  // The password of an existing Mattermost account
```
The // commands make it look quite bad on github. Do you know if there is any way to remove the red background?

opalmer replied:
Latest commit should fix it this.

opalmer commented:
Ok now it should be ready, please take another look. Sorry for the commit noise....we can clean that up on merge by just squashing the commits.
